hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -546,6 +562,39 @@ private HoodieFunctionalIndexDefinition getFunctionalIndexDefinition(String inde
     }
   }
 
+  private Pair<Integer, HoodieData<HoodieRecord>> initializeSecondaryIndexPartition() throws IOException {
+    HoodieMetadataFileSystemView fsView = new HoodieMetadataFileSystemView(dataMetaClient, dataMetaClient.getActiveTimeline(), metadata);
+    // Collect the list of latest file slices present in each partition
+    List<String> partitions = metadata.getAllPartitionPaths();
+    fsView.loadAllPartitions();
+
+    List<Pair<String, FileSlice>> partitionFileSlicePairs = new ArrayList<>();
+    partitions.forEach(partition -> {
```

Comment:
would n't this take a long time, if done on the driver?

Reply:
This is actually borrowed from how the existing indexes/partitions are built in metadata table (like RLI partion in `initializeRecordIndexPartition()`)